src/Tools/Source/FakeSign/Program.cs

```diff
@@ -81,9 +81,12 @@ private static bool Validate(PEReader peReader)
             }
 
             CorHeader header = peReader.PEHeaders.CorHeader;
-            if ((header.Flags & CorFlags.StrongNameSigned) == CorFlags.StrongNameSigned)
+            var expectedStrongNameFlag = unSign ? CorFlags.StrongNameSigned : 0;
+            var actualStrongNameFlag = header.Flags & CorFlags.StrongNameSigned;
+
+            if (expectedStrongNameFlag != actualStrongNameFlag)
             {
-                Console.Error.WriteLine("PE file is already strong-name signed.");
+                Console.Error.WriteLine($"PE file is {(unSign ? "not" : "already")} strong-name signed.");
                 return false;
```

Comment:
I see. So we should use -f in our builds then. It may happen that a library is not recompiled and FakSign is executed on the binary that was already there and thus signed by the previous build.